src/App.tsx

```diff
@@ -1,29 +1,54 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+  const items = getNumbers(1, 42).map(n => `Item ${n}`);
```

Comment:
Consider memoizing the 'items' array using React's useMemo hook to prevent unnecessary recalculations on every render.